File: app/src/main/java/com/punchlist/app/data/model/Comment.kt
package com.punchlist.app.data.model

import com.google.firebase.Timestamp

data class Comment(
    val id: String = "",
    val itemId: String = "",
    val userId: String = "",
    val userName: String = "",
    val text: String = "",
    val createdAt: Timestamp? = null,
    val photoUrl: String = ""
) {
    fun toFirestoreMap(): Map<String, Any?> = mapOf(
        "itemId" to itemId,
        "userId" to userId,
        "userName" to userName,
        "text" to text,
        "createdAt" to (createdAt ?: Timestamp.now()),
        "photoUrl" to photoUrl
    )

    companion object {
        fun fromMap(id: String, map: Map<String, Any?>): Comment = Comment(
            id = id,
            itemId = map["itemId"] as? String ?: "",
            userId = map["userId"] as? String ?: "",
            userName = map["userName"] as? String ?: "",
            text = map["text"] as? String ?: "",
            createdAt = map["createdAt"] as? Timestamp,
            photoUrl = map["photoUrl"] as? String ?: ""
        )
    }
}

File: PunchlistApp/Models/Comment.cs
namespace PunchlistApp.Models;

public class Comment
{
    public string Id { get; set; } = "";
    public string ProjectId { get; set; } = "";
    public string ItemId { get; set; } = "";
    public string UserId { get; set; } = "";
    public string UserName { get; set; } = "";
    public string Text { get; set; } = "";
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public string PhotoUrl { get; set; } = "";
}
